### PR TITLE
feat: i18n

### LINKED ISSUE #196: Implement i18n in Login
- [ ] Whole new login has i18n translations
- [ ] en is available
- [ ] de is available
- [ ] error messages are translated, no backend errors

### PR DESCRIPTION
**Pull request by peintnermax**
Closes  #196

This PR implements i18n. The users language is determined based on the language header and is overwritten by setting a http only cookie. The language can be set from a dropdown in the footer.
This implementation provides the following languages: `en`, `de`, `it`, `es`.
If a key is not found in a certain language, it falls back to the english translation file.

TODOs:
- [ ] Error messages
- [ ] Form labels
- [x] Fix Unit Tests

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] Vitest unit tests ensure that components produce expected outputs on different inputs.
- [ ] Cypress integration tests ensure that login app pages work as expected. The ZITADEL API is mocked.
- [x] No debug or dead code
- [x] My code has no repetitions

### CODE REVIEW

File: apps/login/locales/en.json [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  "verify": {
+    "title": "Verify user",
+    "description": "Enter the Code provided in the verification email.",
+    "userIdMissing": "No userId provided!",
+    "resendCode": "Resend code",
+    "submit": "Continue"
+  },
+  "error": {
+    "unknownContext": "Could not get the context of the user. Make sure to enter the username first or provide a loginName as searchParam.",
+    "sessionExpired": "You need to have a valid session in order to continue."
```
Comment (eliobischof) [inline apps/login/locales/en.json:137]: Write that the session is expired?

File: apps/login/src/app/(login)/idp/[provider]/failure/page.tsx [hunk 2]
```diff
@@ -18,36 +19,19 @@ export default async function Page({
   searchParams: Record<string | number | symbol, string | undefined>;
   params: { provider: string };
 }) {
-  const { id, token, authRequestId, organization } = searchParams;
```
Comment (eliobischof) [inline apps/login/src/app/(login)/idp/[provider]/failure/page.tsx:21]: Why is this removed?

File: apps/login/src/app/(login)/idp/[provider]/success/page.tsx [hunk 3]
```diff
@@ -11,191 +11,158 @@ import {
...
-              </div>
-            </DynamicTheme>
-          );
-        }
-      })
-      .catch((error) => {
+
+  if (!provider || !id || !token) {
+    return loginFailed(branding);
+  }
```
Comment (eliobischof) [inline apps/login/src/app/(login)/idp/[provider]/success/page.tsx:48]: :kissing_heart:

File: apps/login/src/app/(login)/idp/[provider]/success/page.tsx [hunk 4]
```diff
@@ -11,191 +11,158 @@ import {
...
+
+  const intent = await retrieveIDPIntent(id, token);
+
+  const { idpInformation, userId } = intent;
+
+  if (userId && id && token) {
```
Comment (eliobischof) [inline apps/login/src/app/(login)/idp/[provider]/success/page.tsx:54]: id and toke are never falsy, see https://github.com/zitadel/typescript/pull/217/files#r1797378651

File: apps/login/src/components/password-form.tsx [hunk 5]
```diff
@@ -67,6 +70,7 @@ export function PasswordForm({
 
     setLoading(false);
 
+    console.log(response);
```
Comment (eliobischof) [inline apps/login/src/components/password-form.tsx:73]: is this intended?

File: apps/login/src/components/password-form.tsx [hunk 6]
```diff
@@ -108,6 +112,11 @@ export function PasswordForm({
       !submitted.authMethods ||
       !submitted.factors?.user?.loginName
     ) {
+      console.log(
+        !submitted,
+        !submitted?.authMethods,
+        !submitted?.factors?.user?.loginName,
+      );
```
Comment (eliobischof) [inline apps/login/src/components/password-form.tsx:119]: Is this intended?

File: apps/login/src/components/password-form.tsx [hunk 7]
```diff
@@ -117,6 +126,7 @@ export function PasswordForm({
         m !== AuthenticationMethodType.PASSKEY,
     );
 
+    console.log(availableSecondFactors);
```
Comment (eliobischof) [inline apps/login/src/components/password-form.tsx:129]: Is this intended?

File: apps/login/src/components/password-form.tsx (hunk 6 above)
Comment (peintnermax) [reply apps/login/src/components/password-form.tsx]: No logs are removed now

File: apps/login/src/app/(login)/idp/[provider]/success/page.tsx (hunk 4 above)
Comment (peintnermax) [reply apps/login/src/app/(login)/idp/[provider]/success/page.tsx:54]: yeah, but these are not from the api, these are the search params from the url which can be undefined so we have to add this check

File: apps/login/locales/en.json (hunk 1 above)
Comment (peintnermax) [reply apps/login/locales/en.json]: I've changed the texts now

File: apps/login/src/components/password-form.tsx (hunk 7 above)
Comment (peintnermax) [reply apps/login/src/components/password-form.tsx]: removed

File: apps/login/src/app/(login)/idp/[provider]/success/page.tsx (hunk 4 above)
Comment (eliobischof) [reply apps/login/src/app/(login)/idp/[provider]/success/page.tsx:54]: ?

```suggestion
  if (userId) {
```

**Review by eliobischof — APPROVED** [review]

File: apps/login/src/app/(login)/idp/[provider]/success/page.tsx (hunk 4 above)
Comment (peintnermax) [reply apps/login/src/app/(login)/idp/[provider]/success/page.tsx]: ah now I get it.. I've removed the redundant check.. thx 👍

**Review by eliobischof — APPROVED** [review]

**Pull request merged**